proposals/NNNN-same-app-from-multiple-devices.md

```diff
@@ -0,0 +1,80 @@
+
+- SDL shall treat the same app from different connected devices as different app instances or different apps. 
+- SDL shall treat the same app from the same device with multiple transport connections as the same app instance. 
+- SDL shall allow only one instance of the app from one device. I.e. SDL shall reject the app with same app name or the same app id from one device as before.
+- As before, SDL assigns unique internal app id to each app instance.
+- It is up to HMI to differentiate the same app across different devices, for example each device has its own list of apps on different pages, or HMI shows app name together with device name, or HMI appends a number/letter at the end of the `appName`. It is out of the scope of this proposal. 
+
+Mobile API update.
+There is no API change. Just update some descriptions.
+```xml
+<param name="appName" type="String" maxlength="100" mandatory="true" since="1.0">
+    <description>
+        The mobile application name, e.g. "Ford Drive Green".
+-        Needs to be unique over all applications.
++        Needs to be unique over all applications from the same device.
+        May not be empty.
+        May not start with a new line character.
+        May not interfere with any name or synonym of previously registered applications and any predefined blacklist of words (global commands)
+-        Needs to be unique over all applications. Applications with the same name will be rejected.
++        Additional applications with the same name from the same device will be rejected.
+       Only characters from char set [@TODO: Create char set (character/hex value) for each ACM and refer to] are supported.
+    </description>
+</param>
+```
+
+HMI API update
+```xml
+<param name="appName" type="String" maxlength="100" mandatory="false">
+    <description>
+        Request new app name registration
+-        Needs to be unique over all applications.
++        Needs to be unique over all applications from the same device.
+        May not be empty. May not start with a new line character.
+        May not interfere with any name or synonym of any registered applications.
+-        Applications with the same name will be rejected. (SDL makes all the checks)
++        Additional applications with the same name from the same device will be rejected.
+    </description>
+</param>
+```
+## Potential downsides
+
+None.
+
+## Impact on existing code
+
+SDL core code need updates. Currently, SDL only allows one instance of the same app to register with SDL. When an app registers app interface, SDL checks if there is any existing app that has the same `app_policy_id`, and rejects the request if there is one. This need a change. There might be other code changes not aware by the author in order to support this feature.
```

Comment:
Change `This need a change` to `This needs a change.`

Reply:
changed